File: algorithms/2_medium/M/04034_minimum_bishop_moves_to_reach_target/solution.c
int minBishopMoves(const int *source, const int source_len, const int *target, const int target_len) {
    const int sx = source[0];
    const int sy = source[1];
    const int tx = target[0];
    const int ty = target[1];

    if (sx == tx && sy == ty) {
        return 0;
    }

    if (((sx + sy) & 1) != ((tx + ty) & 1)) {
        return -1;
    }

    if (ABS(tx - sx) == ABS(ty - sy)) {
        return 1;
    }

    return 2;
}

File: algorithms/2_medium/M/04034_minimum_bishop_moves_to_reach_target/solution.rs
impl Solution {
    pub fn min_bishop_moves(source: Vec<i32>, target: Vec<i32>) -> i32 {
        let (sx, sy) = (source[0], source[1]);
        let (tx, ty) = (target[0], target[1]);

        if sx == tx && sy == ty {
            return 0;
        }

        if ((sx + sy) & 1) != ((tx + ty) & 1) {
            return -1;
        }

        if (tx - sx).abs() == (ty - sy).abs() {
            return 1;
        }

        2
    }
}
